### PR TITLE
Tx.router.extended.base

### LINKED ISSUE #27: Add Transit Router Enrollment

### PR DESCRIPTION
**Pull request by andrewpmartinez**
- consolidates enrollment logic to use the same pattern (i.e. edge routers are no longer unique)
- adds transit router support to the Edge API
- adds transit router enrollment to the Edge API, resolves #27 
- updates Edge API tests to inspect and validate CRUD responses for proper envelope format/parsing
- adds transit router create/enroll/delete tests

### CODE REVIEW

File: controller/persistence/migrations.go [hunk 1]
```diff
@@ -63,10 +63,14 @@ func (m *Migrations) migrate(step *boltz.MigrationStep) int {
 		return 4
 	}
 
-	// current version
 	if step.CurrentVersion == 4 {
```
Comment (plorenz) [inline controller/persistence/migrations.go:66]: This should return 5

File: controller/persistence/transit_router_store.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+	store.indexName = store.addUniqueNameField()
+	store.symbolEnrollments = store.AddFkSetSymbol(FieldTransitRouterEnrollments, store.stores.enrollment)
+}
+
+func (store *transitRouterStoreImpl) initializeLinked() {
+
+	store.AddLinkCollection(store.symbolEnrollments, store.stores.enrollment.symbolIdentity)
+
+	//todo: confirm this is needed
```
Comment (plorenz) [inline controller/persistence/transit_router_store.go:130]: I don't think this is needed. Should delete the base store bucket which will clear everything underneath.

File: controller/persistence/migrations.go (hunk 1 above)
Comment (andrewpmartinez) [reply controller/persistence/migrations.go:66]: Fixed

File: controller/persistence/transit_router_store.go (hunk 2 above)
Comment (andrewpmartinez) [reply controller/persistence/transit_router_store.go]: Fixed

**Pull request merged**